src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,121 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final RegistrationService registrationService = new RegistrationServiceImpl();
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 5;
+    private static final int MIN_PASSWORD_LENGTH = 5;
+    private static final String LOGIN_WITH_MIN_LENGTH = "io0p1";
+    private static final String PASSWORD_WITH_MIN_LENGTH = "12345";
+    private final StorageDao storageDao = new StorageDaoImpl();
```

Comment:
Avoid using the `StorageDao` directly in tests for the `RegistrationService`. Instead, use mocking to simulate the storage behavior. This will make the tests more reliable and independent of the actual storage implementation.